### PR TITLE
Breaking pixeldataset.py up into separate files

### PR DESCRIPTION
**Pull request by johandahlberg**
## Description

This separate the huge `pixeldataset.py` file into separate smaller files. I suggest focusing on if you think that the module names (and associated documentation) make sense when reviewing this, since nothing else has really been changed.

Fixes: EXE-1078

## Type of change

Purely refactoring.

## How Has This Been Tested?

Ran unit tests locally.

### CODE REVIEW

File: src/pixelator/annotate/__init__.py [hunk 1]
```diff
@@ -1,6 +1,4 @@
-"""
-This module contains functions for the filtering and annotation of
-pixel data in edge list format.
+"""Functions for filtering and annotating of pixel data in edge list format.
```
Comment (ambarrio) [inline src/pixelator/annotate/__init__.py:1]: pixel -> MPX

File: src/pixelator/pixeldataset/__init__.py [hunk 2]
```diff
@@ -0,0 +1,400 @@
...
+
+    @property
+    def adata(self) -> AnnData:
+        """Get the AnnData object."""
+        return self._backend.adata
+
+    @adata.setter
+    def adata(self, value: AnnData) -> None:
+        """Set the AnnData object."""
+        self._backend.adata = value
```
Comment (ambarrio) [inline src/pixelator/pixeldataset/__init__.py:150]: Nice add-on with codecov!

File: src/pixelator/pixeldataset/__init__.py [hunk 3]
```diff
@@ -0,0 +1,400 @@
...
+        """Get the metadata dictionary."""
+        metadata = self._backend.metadata
+        if not metadata:
+            return {}
+        return metadata
+
+    @metadata.setter
+    def metadata(self, value: Dict) -> None:
+        """Set the metadata object."""
+        self._backend.metadata = value
```
Comment (ambarrio) [inline src/pixelator/pixeldataset/__init__.py:205]: It seems clear we are not testing the setters.

File: src/pixelator/pixeldataset/aggregation.py [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+
+def simple_aggregate(
+    sample_names: List[str],
+    datasets: List[PixelDataset],
+    ignore_edgelists: bool = False,
+) -> PixelDataset:
+    """Aggregate samples in a simple way (see caveats).
+
+    Aggregating samples in a simplistic fashion. This function should only
+    be used if the dataset you merge have been generated with the same panel.
```
Comment (ambarrio) [inline src/pixelator/pixeldataset/aggregation.py:68]: Is there anyway we could assert for this in the function? I think it should be possible as we store the panel and could easily check the metadata or the targets.

**Review by ambarrio — COMMENTED** [review]
See my comments and the codecov reports.

File: src/pixelator/pixeldataset/__init__.py (hunk 3 above)
Comment (johandahlberg) [reply src/pixelator/pixeldataset/__init__.py:205]: Yeah. However, I suggest we stick to only the refactoring in this PR, and go on with improving test coverage later. I think it's easier when large scale moves (as this one) are not conflated with adding/removing code.

File: src/pixelator/pixeldataset/aggregation.py (hunk 4 above)
Comment (johandahlberg) [reply src/pixelator/pixeldataset/aggregation.py:68]: Right now we require that the antibodies are the same, but we probably want to do a more advanced aggregation here. Perhaps you want to add a ticket about that and we can follow up on that?

**Review by fbdtemme — APPROVED** [review]

**Pull request merged**